src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PaymentForExistingCaseTest.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.hmcts.reform.bulkscan.payment.processor;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.ccd.CcdAuthenticator;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.ccd.CcdAuthenticatorFactory;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.helper.CaseSearcher;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.helper.ExceptionRecordCreator;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.helper.PaymentsMessageSender;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.model.CreatePaymentsCommand;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.model.PaymentData;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+@SpringBootTest
+@ActiveProfiles("functional")
+class PaymentForExistingCaseTest {
```

Comment:
This class contains a test that checks if CCD was properly updated after payment creation was requested through the queue. I'd suggest a different name, e.g. `CcdUpdateTest` or more general `PaymentProcessingTest`.

Reply:
This test is testing PaymentProcessor, not CCD, so CcdUpdateTest is not the best class name. 
This test is testing payment sent for a case which already exists, there are no other test cases in it, so its name is quite precise at the moment. There is no indication so far that more tests will be included in this class. My choice is to keep existing name.

Reply:
How about `PaymentForExceptionRecordTest` then? Very shortly we're going to introduce payment creation for an existing case, not exception record. There will be a confusion with this name.